Fix NoMethodError when using `should_not allow_value` with Minitest+shoulda-context

mattbrictson opened the pull request:
When using Minitest (i.e. `ActiveSupport::TestCase`) in combination with shoulda-context, a failing `should_not allow_value...` assertion would result in this error:

```
NoMethodError: undefined method `negative_failure_message'
```

This commit fixes this error by defining `failure_message_for_should_not`, which shoulda-context will use instead.

mcmire commented:
Ah... I had no clue.

Do you think this is something we should change in shoulda-context instead?

mattbrictson commented:
> Do you think this is something we should change in shoulda-context instead?

Meaning, perhaps we should change shoulda-context to use `failure_message_when_negated` instead? I am just barely familiar with the code, so I don't know whether that would be a high-risk change. My goal with this PR was to make the simplest change that could fix the bug.

If you think it's better to change shoulda-context, I can create a PR there instead.

mcmire commented:
Alright. Let me think about this a little more and get back to you.

mattbrictson commented:
I rebased against current `master` and fixed the conflicts.

I'm still curious if you are open to merge this or if should pursue changes to shoulda-context instead.

mcmire commented:
I'm about to go on a long trip overseas, and I'm afraid I don't have time to address this right now.

I do think this change should be made in shoulda-context, though, simply because it'll be easier. I believe in order to make this work here, you'd have to add `failure_message_for_should_not` to every single matcher, not just `allow_value`, which I'd want to avoid.

mcmire commented:
There, I even added a tag :)

mcmire commented:
Hey folks. In an effort to lighten our load as maintainers and be able to serve you better in the future, the shoulda-matchers team is working on cleaning out the cobwebs in this repo by pruning the backlog. As there are few of us, there are a lot of items that will simply never earn our attention in a reasonable time frame, and rather than giving you an empty promise, we think it makes more sense to focus on more recent issues. That means, unfortunately, that we must close this PR.

Don't take this the wrong way: our aim is not to diminish the effort people have made or dismiss problems that have been raised. If you feel that we should reopen this PR, then please let us know so that we can reprioritize it. Thanks!
